src/main/java/org/junitpioneer/jupiter/issue/IssuedTestCase.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter.issue;
+
+/**
+ * Represents the execution result of test method, which is annotated with {@link org.junitpioneer.jupiter.Issue}.
+ *
+ * In future java this could be a record.
+ */
+public final class IssuedTestCase {
+
+	private final String uniqueName;
+	private final String issueId;
+	private final String result;
```

Comment:
I don't agree :D

Reply:
I wanted to recommend `TestExecutionResult` as well, but it doesn't know the test's id, so we have to store that ourselves anyway. Instead of referencing it, we can just repack the information we need into our own class - see my proposal for `IssueTestCase` above.